shotover/src/message/mod.rs

```diff
@@ -311,10 +318,10 @@ impl Message {
         match self.inner.unwrap() {
             MessageInner::RawBytes { bytes, .. } => Encodable::Bytes(bytes),
             MessageInner::Parsed { bytes, .. } => Encodable::Bytes(bytes),
-            MessageInner::Modified {
-                frame: Frame::Dummy,
-            } => Encodable::Bytes(Bytes::new()),
-            MessageInner::Modified { frame } => Encodable::Frame(frame),
+            MessageInner::Modified { frame } => match *frame {
+                Frame::Dummy => Encodable::Bytes(Bytes::new()),
+                frame => Encodable::Frame(Box::new(frame)),
```

Comment:
Looking good! one last thing to fix.

we unbox and rebox the frame here which is expensive.
Instead we can access the frame as a reference to check if its a dummy, and then move the original boxed frame into `Encodeable::Frame`.

Like this:
![image](https://github.com/user-attachments/assets/06e6f07e-df1c-4861-84f8-52a3fdbd0add)